packages/system/src/system.ts

```diff
@@ -18,7 +18,6 @@ import emotionStyled, {
   FunctionInterpolation,
   Interpolation,
 } from "@emotion/styled"
-import { shouldForwardProp } from "./should-forward-prop"
```

Comment:
I think the goal here is not to remove the `shouldForwardProp` function but to apply other custom `shouldForwardProp` passed in the `options` **after** the default `shouldForwardProp` has been called.

The logic is more like:

```jsx
const shouldForwardPropFn = (prop) => {
  // if prop is listed in our styled system, use chakra's shouldForwardProp
  // else if options.shouldForwardProp is defined, use that
 // else return `true`
}
```

Reply:
OK, I understand this right now. I will fix today.

Reply:
imo I think what we ultimately want is to merge the two together into a single function.

If `options.shouldForwardProp` does not exist: use the existing `shouldForwardProp` function

If `options.shouldForwardProp` does exist: create a new function that takes `prop`, calls `shouldForwardProp` and `options.shouldForwardProp`, and then merges the results together. 

That way `options.shouldForwardProp` becomes an _extension_ of the base `shouldForwardProp` function rather than a replacement. 

Not sure that change makes sense for this PR or if it should be separate work.

Reply:
@with-heart and @segunadebayo pushed some changes.

Reply:
not finished, this part not makes senses to me

If options.shouldForwardProp does exist: create a new function that takes prop, calls shouldForwardProp and options.shouldForwardProp, and then merges the results together.

I don't have the props in this scope, how can I merge the two in one? I have no idea.

Reply:
We create a new function. And this new function will be called with the prop name, and we call the existing functions to compute the result.

Define a new function which receives any arguments and pass them to the two different `shouldForwardProps` functions. If one of those functions return `true`, the props will be passed through.

An implementation could look like this:

```ts
import { shouldForwardProp as shouldForwardPropFn } from './should-forward-prop';

// ...

const mergedShouldForwardPropFn = (...args: any[]) => {
  const origShouldForward = shouldForwardPropFn(...args); // our original fn
  if (!options?.shouldForwardProp) {
    // no custom fn provided
    return origShouldForward;
  }

  const optionShouldForward = options.shouldForwardProp(...args);
  // merge results
  return optionShouldForward || origShouldForward;
};

const opts = { ...styledOptions, shouldForwardProp: mergedShouldForwardPropFn };
// ...
```

(!) This implementation would prevent the `options.shouldForwardProp` from overriding a truthy return type of our `shouldForwardProp` function. I don't know if this is the intended behaviour for this use case. I am not very familiar with this area of chakra yet.

Reply:
> I don't have the props in this scope

It's sort of like how `shouldForwardProps` is defined currently. It's a function that takes `prop`, but we're not actually passing props to it, we're handing the function to `emotionStyled` for it to call it later.

@TimKolberger I think this is close, but it's missing something. I'm going to make a new issue for this so we can discuss a little more. We might need a separate option to handle the use cases I'm thinking of.

@jvmazagao Sorry for starting this conversation in the middle of your PR lol. I think what would resolve #2670 would be to go back to how things worked originally, but `opts` should be `{ shouldForwardProp, ...styledOptions }` instead of `{ ...styledOptions, shouldForwardProp }`, so that if a user includes `styledOptions.shouldForwardProps`, it overrides the built-in one.

Reply:
I see where you're going @with-heart. I think a better solution will be to do this:
`{ shouldForwardProp, ...styledOptions }` and then export our `shouldForwardProp` function so that users can leverage that when they need to build their own custom function.

Similar to how emotion has `@emotion/is-prop-valid` pkg but for us, we're simply exporting it out. This is a simpler approach, nice thought @with-heart!

Reply:
Sorry guys, but I'm a little confused with this discussion, and I love to work in another pr to fix another issue if I'm capable of that.